### ISSUE TITLE
Tmp files deleted before finished extracting

### ISSUE THREAD

**Issue by watchingvideos845** [AchetaGames/Epic-Asset-Manager#244]
**Describe the bug**
Trying to download [HackNSlashV3](https://www.unrealengine.com/marketplace/en-US/item/a54a32a1f168453780954296a9546aac) to play around with. It has 868 files but only 861 is extracted because all temp files are deleted before extraction is finished. Making it impossible to download.

**To Reproduce**
1. Download HackNSlashV3
2. Download finishes
3. Only 861 files are extracted and exist, missing 7 files
4. Can't redownload missing files or anything

**Expected behavior**
All 868 files are extracted. No missing files.

**Screenshots**
![Screenshot_20230920_154258-1](https://github.com/AchetaGames/Epic-Asset-Manager/assets/141391442/52cfcc32-1520-4394-b1c5-f1d7d7a0a587)

**Desktop (please complete the following information):**
 - OS: Linux
 - Browser Firefox
 - Version 117

**Comment by aknarts** [comment]
That is strange. Does it get stuck on the "Extraction" and never finishes? How did you figure the amount of files?

The way it works is there is a chunk list and a file list kind of linked together, a chunk can contain multiple files or pieces of files and only after all of them are read this should get deleted. I will take a look if there is a hole there that removes it before. I can also just redownload the chunk if it is missing at the time of extraction which would be a workaround, though not a pretty one as that would mean redownloading things.

**Comment by watchingvideos845** [comment]
>That is strange. Does it get stuck on the "Extraction" and never finishes? How did you figure the amount of files?

Yes it gets stuck. Sorry I had difficulty showing it but if you hover your mouse on the extraction bar it shows the amount in X/X so it gets stuck on 861/868

>The way it works is there is a chunk list and a file list kind of linked together, a chunk can contain multiple files or pieces of files and only after all of them are read this should get deleted. I will take a look if there is a hole there that removes it before. I can also just redownload the chunk if it is missing at the time of extraction which would be a workaround, though not a pretty one as that would mean redownloading things.

I tried redownloading but the download never starts as it shows that everything is installed and completed. I only have an issue with this one asset so far, the others I have seem to download okay.

**Comment by aknarts** [comment]
Would you be able to send me the asset manifest? Depending on where you get EAM from that gives the location... figuring out the ID is the hard part :)

It is somewhere in `~/.var/app/io.github.achetagames.epic_asset_manager/cache/epic_asset_manager` for flatpak or `~/.cache/epic_asset_manager/`

Sadly I do not think I currently cache the download manifest, which would contain the actual download links.. but maybe this does give me some idea.

**Comment by watchingvideos845** [comment]
Sure thing.

[a54a32a1f168453780954296a9546aac.zip](https://github.com/AchetaGames/Epic-Asset-Manager/files/12724431/a54a32a1f168453780954296a9546aac.zip)
